spec/lib/iam_ssoe_oauth/service_spec.rb

```diff
@@ -7,9 +7,9 @@
   let(:service) { IAMSSOeOAuth::Service.new }
 
   before do
-    allow(IAMSSOeOAuth::Configuration.instance).to receive_messages(
-      ssl_cert: instance_double('OpenSSL::X509::Certificate'), ssl_key: instance_double('OpenSSL::PKey::RSA')
-    )
+    config = IAMSSOeOAuth::Configuration.instance
+    allow(config).to receive_messages(ssl_cert: OpenStruct.new(to_der: 'dummy_cert_data'),
+                                      ssl_key: OpenStruct.new(to_der: 'dummy_key_data'))
```

Comment:
Failed specs in `aim_ssoe_oauth` all had the following error message:

```
#<InstanceDouble(OpenSSL::X509::Certificate) (anonymous)> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for.
```

Using an `OpenStruct` object the the `to_der` method allows us to continue to mock the functionality without having to worry about the limitations of double. Note that I tried every combination of redefining the doubles to get this to work and always received the same errors.